user/test/com/google/gwt/emultest/java/util/ObjectsTest.java

```diff
@@ -102,4 +102,69 @@ public void testHashCode() {
     Object obj = new Object();
     assertEquals(obj.hashCode(), Objects.hashCode(obj));
   }
+
+  public void testRequireNonNull() {
+    assertEquals("foo", Objects.requireNonNull(hideFromCompiler("foo")));
+    assertThrows(NullPointerException.class,
+        () -> Objects.requireNonNull(hideFromCompiler(null)));
+  }
+
+  public void testRequireNonNullElse() {
+    assertEquals("foo", Objects.requireNonNullElse(hideFromCompiler("foo"), "bar"));
+    assertThrows(NullPointerException.class,
+        () -> Objects.requireNonNullElse(hideFromCompiler(null), null));
+  }
+
+  public void testRequireNonNullElseGet() {
+    assertEquals("foo",
+        Objects.requireNonNullElseGet(hideFromCompiler("foo"), () -> "bar"));
+    assertThrows(NullPointerException.class,
+        () -> Objects.requireNonNullElseGet(hideFromCompiler(null), null));
+    assertThrows(NullPointerException.class,
+        () -> Objects.requireNonNullElseGet(hideFromCompiler(null), () -> null));
+  }
+
+  private String hideFromCompiler(String value) {
+    if (Math.random() > 2) {
+      return "unreachable";
+    }
+    return value;
+  }
+
+  public void testCheckIndex() {
+    assertEquals(5, Objects.checkIndex(5, 10));
+    assertThrows(IndexOutOfBoundsException.class,
+        () -> Objects.checkIndex(-5, 5));
+    assertThrows(IndexOutOfBoundsException.class,
+        () -> Objects.checkIndex(10, 5));
+  }
+
+  public void testCheckFromToIndex() {
+    assertEquals(5, Objects.checkFromToIndex(5, 7, 10));
+    assertThrows(IndexOutOfBoundsException.class,
+        () -> Objects.checkFromToIndex(-5, 1, 5));
+    assertThrows(IndexOutOfBoundsException.class,
+        () -> Objects.checkFromToIndex(10, 1,  5));
+    assertThrows(IndexOutOfBoundsException.class,
+        () -> Objects.checkFromToIndex(1, 10,  5));
+  }
+
+  public void testCheckFromIndexSize() {
```

Comment:
Check that 0,10,10 works to ensure that fromIndex + size and length are both exclusive. Also check that size cannot be negative while all other conditions would be fine, e.g. 10, -5, 10.